tools/executor/util.js

```diff
@@ -0,0 +1,60 @@
+const execaRaw = require('execa');
+const AWS = require('aws-sdk');
+
+const AWS_REGION = 'us-east-2';
+
+AWS.config.update({ region: AWS_REGION });
+
+/**
+ * Wrapper around `execa` that logs commands as they're run.
+ * 
+ * @param {string} file 
+ * @param {string[]} args 
+ * @param {import('execa').Options} [options]
+ */
+const execa = (file, args, options) => {
+  let command = file;
+  if (args) {
+    command += ` ${args.join(' ')}`;
+  }
+  console.log(`$ ${command}`);
+  return execaRaw(file, args, options);
+};
+
+const getImageTag = async () => {
+  return (await execaRaw('git', ['rev-parse', 'HEAD'])).stdout;
+};
+
+const getImageName = (tag) => {
+  return `prairielearn/executor:${tag}`;
+};
+
+const getEcrRegistryUrl = async () => {
+  // ECR registries are tied to account IDs. Determine the account ID dynamically
+  // based on the credentials we're running with.
+  const sts = new AWS.STS();
+  const { Account: accountId } = await sts.getCallerIdentity().promise();
```

Comment:
Does this require running on an EC2 instance, or will it pull AWS config from the current environment?

Reply:
This uses the standard hierarchy of credentials - environment variables, shared credentials file, EC2 metadata service, etc.